index.js

```diff
@@ -30,12 +35,27 @@ debug({ showDevTools: false })
 // Handle uncaught errors
 // XXX(KM): why aren't we enabling this?
 // catchErrors({ onError: handleError })
+var pid = new Pid(userDataPath)
+
+// Before we do anything, let's make sure we're ready to gracefully shut down
+var serverProcess = null
+const onExit = require('capture-exit')
+const signalExit = require('signal-exit')
+onExit.captureExit()
+onExit.onExit(() => {
+  killZombies()
+})
+
+signalExit((code, signal) => {
```

Comment:
@noffle I don't know if we need both of these at this point, capture-exit works for graceful exits during async calls and signal-exit is for external sig* sends...

Reply:
Oh, if signal-exit doesn't fire on graceful exits, then we'd need both to capture graceful & nongraceful exits, right?

Reply:
yeah. in any case, they will only ever be called once!